### PR TITLE
Upgrade Graal 23 to 23.0.1

### PR DESCRIPTION
**Pull request by keeganwitt**

### CODE REVIEW

**Comment by tianon** [comment]
Doh, I didn't notice it before, but `gradle:jdk-21-and-22-graal-jammy` now contains 21 and _23_, right? :sweat_smile:  Is that intentional?

**Comment by keeganwitt** [comment]
> Doh, I didn't notice it before, but `gradle:jdk-21-and-22-graal-jammy` now contains 21 and _23_, right? 😅 Is that intentional?

Oops, no, that should have been changed. Added to this MR.

File: library/gradle [hunk 1]
```diff
@@ -6,92 +6,92 @@ GitRepo: https://github.com/keeganwitt/docker-gradle.git
...
+GitCommit: 1d17a9a8ffeac966db523f07dc8fa79dfc30a0f9
 Directory: jdk23-alpine
 
 Tags: 8.10.2-jdk23-graal, 8.10-jdk23-graal, 8-jdk23-graal, jdk23-graal
 Architectures: amd64, arm64v8
-GitCommit: fbc29c4f1b70066178a708fffdd1c781efaa8f5a
+GitCommit: 1d17a9a8ffeac966db523f07dc8fa79dfc30a0f9
 Directory: jdk23-graal
 
 Tags: 8.10.2-jdk-lts-and-current, 8.10-jdk-lts-and-current, 8-jdk-lts-and-current, jdk-lts-and-current, 8.10.2-jdk-lts-and-current-jammy, 8.10-jdk-lts-and-current-jammy, 8-jdk-lts-and-current-jammy, jdk-lts-and-current-jammy,8.10.2-jdk-21-and-22, 8.10-jdk-21-and-22, 8-jdk-21-and-22, jdk-21-and-22, 8.10.2-jdk-21-and-22-jammy, 8.10-jdk-21-and-22-jammy, 8-jdk-21-and-22-jammy, jdk-21-and-22-jammy
```
Comment (tianon) [inline library/gradle:82]:
```suggestion
Tags: 8.10.2-jdk-lts-and-current, 8.10-jdk-lts-and-current, 8-jdk-lts-and-current, jdk-lts-and-current, 8.10.2-jdk-lts-and-current-jammy, 8.10-jdk-lts-and-current-jammy, 8-jdk-lts-and-current-jammy, jdk-lts-and-current-jammy,8.10.2-jdk-21-and-23, 8.10-jdk-21-and-23, 8-jdk-21-and-23, jdk-21-and-23, 8.10.2-jdk-21-and-23-jammy, 8.10-jdk-21-and-23-jammy, 8-jdk-21-and-23-jammy, jdk-21-and-23-jammy
```

```console
$ docker run -it --rm --pull=always gradle:jdk-21-and-22-jammy ls -lAFh /opt/java/
jdk-21-and-22-jammy: Pulling from library/gradle
Digest: sha256:78e6a7c94a729be925ffd6aa65c14ae8cd6beeced3189da520778fbe245ff3d9
Status: Image is up to date for gradle:jdk-21-and-22-jammy
total 8.0K
drwxr-xr-x 9 root root 4.0K Sep 17 01:03 openjdk/
lrwxrwxrwx 1 root root   17 Oct 16 16:15 openjdk21 -> /opt/java/openjdk/
drwxr-xr-x 9 root root 4.0K Oct 16 02:15 openjdk23/
```

File: library/gradle [hunk 2]
```diff
@@ -6,92 +6,92 @@ GitRepo: https://github.com/keeganwitt/docker-gradle.git
...
 Directory: jdk-lts-and-current
 
-Tags: 8.10.2-jdk-lts-and-current-alpine, 8.10-jdk-lts-and-current-alpine, 8-jdk-lts-and-current-alpine, jdk-lts-and-current-alpine,8.10.2-jdk-21-and-22-alpine, 8.10-jdk-21-and-22-alpine, 8-jdk-21-and-22-alpine, jdk-21-and-22-alpine
+Tags: 8.10.2-jdk-lts-and-current-alpine, 8.10-jdk-lts-and-current-alpine, 8-jdk-lts-and-current-alpine, jdk-lts-and-current-alpine,8.10.2-jdk-21-and-23-alpine, 8.10-jdk-21-and-23-alpine, 8-jdk-21-and-23-alpine, jdk-21-and-23-alpine
 Architectures: amd64, arm64v8
-GitCommit: fbc29c4f1b70066178a708fffdd1c781efaa8f5a
+GitCommit: 1d17a9a8ffeac966db523f07dc8fa79dfc30a0f9
 Directory: jdk-lts-and-current-alpine
 
 Tags: 8.10.2-jdk-lts-and-current-graal, 8.10-jdk-lts-and-current-graal, 8-jdk-lts-and-current-graal, jdk-lts-and-current-graal, 8.10.2-jdk-lts-and-current-graal-jammy, 8.10-jdk-lts-and-current-graal-jammy, 8-jdk-lts-and-current-graal-jammy, jdk-lts-and-current-graal-jammy,8.10.2-jdk-21-and-22-graal, 8.10-jdk-21-and-22-graal, 8-jdk-21-and-22-graal, jdk-21-and-22-graal, 8.10.2-jdk-21-and-22-graal-jammy, 8.10-jdk-21-and-22-graal-jammy, 8-jdk-21-and-22-graal-jammy, jdk-21-and-22-graal-jammy
```
Comment (tianon) [inline library/gradle:92]:
```suggestion
Tags: 8.10.2-jdk-lts-and-current-graal, 8.10-jdk-lts-and-current-graal, 8-jdk-lts-and-current-graal, jdk-lts-and-current-graal, 8.10.2-jdk-lts-and-current-graal-jammy, 8.10-jdk-lts-and-current-graal-jammy, 8-jdk-lts-and-current-graal-jammy, jdk-lts-and-current-graal-jammy,8.10.2-jdk-21-and-23-graal, 8.10-jdk-21-and-23-graal, 8-jdk-21-and-23-graal, jdk-21-and-23-graal, 8.10.2-jdk-21-and-23-graal-jammy, 8.10-jdk-21-and-23-graal-jammy, 8-jdk-21-and-23-graal-jammy, jdk-21-and-23-graal-jammy
```

**Review by tianon — COMMENTED** [review]
:eyes:

**Review by yosifkit — APPROVED** [review]

**Pull request merged**